docs/2.guide/4.recipes/4.sessions-and-authentication.md

```diff
@@ -0,0 +1,403 @@
+
+The `auth-utils` module provides the `requireUserSession` utility function to help make sure that users are logged in and have an active session. We can use this to protect our different endpoints. Like many of the other utilities from the auth module, it is automatically imported in our server endpoints.
+
+In the example below, we use the `requireUserSession` utility function to protect the `/server/api/users.get.ts` server route. This route will only be accessible to logged-in users.
+
+```typescript [/server/api/users.get.ts]
+import getDatabase from "~/database/database";
+import users from "~/database/schema/users";
+import requireUserLoggedIn from "~/server/utils/requireUserLoggedIn";
+
+export default defineEventHandler(async (event) => {
+  // make sure the user is logged in
+  // This will throw a 401 error if the request doesn't come from a valid user session
+  await requireUserSession(event);
+
+  // If we make it hear, the user is authenticated. It's safe to fetch and return data
+  const db = await getDatabase();
+  
+  // Send back the list of users
+  const userList = await db.select({ name: users.name, id: users.id }).from(users).limit(10);
+
+  return userList;
+});
+```
+
+### 8. Create a front-end middleware to protect routes
+
+Our data are safe with the back-end route in place, but without doing anything else, unauthenticated users would probably get some odd data when trying to access the `/users` page. We should create a [front-end middleware](https://nuxt.com/docs/guide/directory-structure/middleware) to protect the route on the client side and redirect users to a login page.
+
+`nuxt-auth-utils` provides a convenient `useUserSession` composable which we'll use to check if the user is actually logged in, and redirect them if they are not.
+
+We'll create a middleware in the `/middleware` directory. Unlike on the server, front-end middleware is not automatically applied to all endpoints, and we'll need to specify where we want it applied.
+
+```typescript [/middleware/RedirectIfNotAuthenticated.ts]
+export default defineNuxtRouteMiddleware(() => {
+    // check if the user is logged in
+    const { loggedIn } = useUserSession();
+
+    // redirect the user to the login screen if they're not authenticated
+    if (!loggedIn.value) {
+        return navigateTo("/login");
+    }
+
+    return null;
+});
+```
+
+### 9. Protect a route with the front-end middleware
+
+Now that we have the front-end middleware to protect front-end routes, we can use it in any page to ensure that only logged-in users can access the route. Users will be redirected to the login page if they are not authenticated.
+
+We'll use [`definePageMeta`](https://nuxt.com/docs/api/utils/define-page-meta) to apply the middleware to the route that we want to protect.
+
+::important
+:warning: Remember that your data aren't really secure without back-end protection! Always secure your data on the back-end first before worrying about the front-end.
```

Comment:
"Data isn't really" 😄 
Also same suggestions about back-end and front-end

Reply:
Fixed, back/front, left is/are for now.